ghcide/src/Development/IDE/Core/Shake.hs

```diff
@@ -259,6 +261,8 @@ data ShakeExtras = ShakeExtras
     ,publishedDiagnostics :: STM.Map NormalizedUri [Diagnostic]
     -- ^ This represents the set of diagnostics that we have published.
     -- Due to debouncing not every change might get published.
+    ,semanticTokensCache:: STM.Map NormalizedUri SemanticTokens
+    ,semanticTokensId :: TVar Int
```

Comment:
Some docs please, why we have them, etc... :)

Reply:
done

Reply:
Thanks, much better!